### PR TITLE
Save panel buffer for images loaded through import panel

### PR DESCRIPTION
**Pull request by bnmajor**
Fix bug that was writing to copy of object rather than original object. Write temp file to `hdf5` rather than `yaml`.

Attached are the images and config files for before processing through the import panel and after.
[data.zip](https://github.com/HEXRD/hexrdgui/files/6881384/data.zip)

### CODE REVIEW

File: hexrd/ui/import_data_panel.py [hunk 1]
```diff
@@ -434,21 +435,22 @@ def completed(self):
...
-        temp.write(data.encode('utf-8'))
-        temp.close()
+        temp = tempfile.NamedTemporaryFile(delete=False, suffix='.hexrd')
+        try:
+            instr = HEDMInstrument(
+                instrument_config=self.detector_defaults['default_config'])
+            instr.write_config(temp.name, style='hdf5')
+            HexrdConfig().load_instrument_config(temp.name)
+        finally:
+            Path(temp.name).unlink()
```
Comment (psavery) [inline hexrd/ui/import_data_panel.py:452]: Do we need to close it before we unlink it? I'm not sure...

**Review by psavery — APPROVED** [review]
LGTM. I did not test it, so I'm relying on @bnmajor to have tested it.

**Comment by joelvbernier** [comment]
gonna test now...

File: hexrd/ui/import_data_panel.py (hunk 1 above)
Comment (psavery) [reply hexrd/ui/import_data_panel.py:452]: Looks like we need to close the file or we will get an error on Windows when we try to unlink it.

Comment (bnmajor) [reply hexrd/ui/import_data_panel.py:454]: Yes, fixed.

**Comment by joelvbernier** [comment]
First, was getting an error relating to the hdf5 format temp file (instrument config?):
```
Traceback (most recent call last):
  File "c:\users\berni\documents\github\hexrdgui\hexrd\ui\import_data_panel.py", line 452, in completed
    Path(temp.name).unlink()
  File "C:\Users\berni\miniconda3\envs\hexrdgui_dev\lib\pathlib.py", line 1321, in unlink
    self._accessor.unlink(self)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\berni\\AppData\\Local\\Temp\\tmpehj5vmr2.hexrd'
```

**Comment by joelvbernier** [comment]
After that fix, I am now getting a error from the YAML load of... a config file?
```
Traceback (most recent call last):
  File "c:\users\berni\documents\github\hexrdgui\hexrd\ui\import_data_panel.py", line 240, in load_images
    success = self.get_instrument_defaults()
  File "c:\users\berni\documents\github\hexrdgui\hexrd\ui\import_data_panel.py", line 98, in get_instrument_defaults
    self.defaults = yaml.load(f, Loader=yaml.FullLoader)
  File "C:\Users\berni\miniconda3\envs\hexrdgui_dev\lib\site-packages\yaml\__init__.py", line 114, in load
    return loader.get_single_data()
  File "C:\Users\berni\miniconda3\envs\hexrdgui_dev\lib\site-packages\yaml\constructor.py", line 51, in get_single_data
    return self.construct_document(node)
  File "C:\Users\berni\miniconda3\envs\hexrdgui_dev\lib\site-packages\yaml\constructor.py", line 60, in construct_document
    for dummy in generator:
  File "C:\Users\berni\miniconda3\envs\hexrdgui_dev\lib\site-packages\yaml\constructor.py", line 413, in construct_yaml_map
    value = self.construct_mapping(node)
  File "C:\Users\berni\miniconda3\envs\hexrdgui_dev\lib\site-packages\yaml\constructor.py", line 218, in construct_mapping
    return super().construct_mapping(node, deep=deep)
  File "C:\Users\berni\miniconda3\envs\hexrdgui_dev\lib\site-packages\yaml\constructor.py", line 143, in construct_mapping
    value = self.construct_object(value_node, deep=deep)
  File "C:\Users\berni\miniconda3\envs\hexrdgui_dev\lib\site-packages\yaml\constructor.py", line 100, in construct_object
    data = constructor(self, node)
  File "C:\Users\berni\miniconda3\envs\hexrdgui_dev\lib\site-packages\yaml\constructor.py", line 427, in construct_undefined
    raise ConstructorError(None, None,
yaml.constructor.ConstructorError: could not determine a constructor for the tag 'tag:yaml.org,2002:python/object/apply:numpy.core.multiarray.scalar'
  in "C:\Users\berni\AppData\Local\Temp\tmp0auwk70j.yml", line 2, column 11
```

**Comment by bnmajor** [comment]
@joelvbernier Are you still seeing the Permission Error? That should have been fixed with commit 21ef39f but if your branch is up to date and you're still seeing the issue please let me know.

The second error that you mentioned came from the fact that the config previously always defaulted to being saved and read in as a `yml` file, but since panel buffers require that the format be `hdf5` this has been updated to default to `hdf5` as well.

File: hexrd/ui/import_data_panel.py [hunk 2]
```diff
@@ -99,8 +99,8 @@ def get_instrument_defaults(self):
             else:
                 try:
                     with h5py.File(self.config_file, 'r') as f:
-                        instr = create_hedm_instrument()
-                        instr.unwrap_h5_to_dict(f, self.defaults)
+                        instr = HEDMInstrument(f)
```
Comment (psavery) [inline hexrd/ui/import_data_panel.py:102]: I'm not 100% sure of everything that is going on here, but you may need to pass in the `tilt_calibration_mapping` similar to what is done [here](https://github.com/HEXRD/hexrdgui/blob/79b0b0cf568392fac413a63a0500ddf8f097c67b/hexrd/ui/hexrd_config.py#L496) via `HexrdConfig().rotation_matrix_euler()`.

File: hexrd/ui/import_data_panel.py [hunk 3]
```diff
@@ -22,7 +23,7 @@
     UI_TRANS_INDEX_ROTATE_90, UI_TRANS_INDEX_FLIP_HORIZONTALLY, YAML_EXTS)
 import hexrd.ui.resources.calibration
 
-from hexrd.ui.utils import convert_tilt_convention
+from hexrd.ui.utils import convert_tilt_convention, instr_to_internal_dict
```
Comment (joelvbernier) [inline hexrd/ui/import_data_panel.py:26]: @bnmajor this is getting the tilt conversion, correct?

Comment (joelvbernier) [reply hexrd/ui/import_data_panel.py:26]: `instr_to_internal_dict` that is.

File: hexrd/ui/import_data_panel.py [hunk 4]
```diff
@@ -427,28 +429,29 @@ def completed(self):
...
+            instrument_config=self.detector_defaults['default_config'])
         for det in self.completed_detectors:
-            transform = detectors[det].setdefault('transform', {})
-            *zx, z = transform['tilt']
-            transform['tilt'] = (
-                [*zx, (z + float(self.edited_images[det]['tilt']))])
-            panel_buffer = detectors[det].setdefault('panel_buffer', [])
-            panel_buffer = self.edited_images[det]['panel_buffer']
+            panel = instr.detectors[det]
+            *zx, z = panel.tilt
```
Comment (joelvbernier) [inline hexrd/ui/import_data_panel.py:442]: you'll need to import the following methods:
```
from hexrd.rotations import angleAxisOfRotMat, make_rmat_euler
```

then replace 442-443 with the following:
```
# first need the zxz Euler angles from the panel rotation matrix.
*zx, z = angles_from_rmat_zxz(panel.rmat)

# convert updated zxz angles to rmat
rmat_updated = make_rmat_euler(
    [*zx, (z + float(self.edited_images[det]['tilt']))],
    'zxz', extrinsic=False
)

# convert to angle-axis parameters
rang, raxs = angleAxisOfRotMat(rmat_updated)

# update tilt property on panel
panel.tilt = rang*raxs.flatten()
```

Comment (bnmajor) [reply hexrd/ui/import_data_panel.py]: This has been added in, thanks for the help!

**Comment by bnmajor** [comment]
@joelvbernier The code changes that you suggested have been added in now and it finally appears that angles are being updated correctly.

One change that I made was to remember the angle of rotation when storing the template position. This was being lost previously and would result in the tilt not reflecting that initial rotation when loading a remembered template. In order to guard against errors or bugs with old configurations though the templates stored in the old format are forgotten now. This means that you will need to re-position everything again (but only once) so that the data is stored in the correct and expected format going forward. Sorry for the inconvenience!

**Review by joelvbernier — APPROVED** [review]

**Pull request merged**